test/e2e/backup_test.go

```diff
@@ -87,3 +105,62 @@ func TestEtcdBackupOperatorForS3Backup(t *testing.T) {
 		t.Fatalf("failed to verify backup: %v", err)
 	}
 }
+
+// testEtcdRestoreOperatorForS3Source tests if the restore-operator can restore an etcd cluster from an S3 restore source
+func testEtcdRestoreOperatorForS3Source(t *testing.T) {
+	if os.Getenv(envParallelTest) == envParallelTestTrue {
+		t.Parallel()
+	}
+	f := framework.Global
+
+	awsSecret := os.Getenv("TEST_AWS_SECRET")
+	er := e2eutil.NewS3EtcdRestore("test-etcd-restore-", "3.1.8", TestS3Path, awsSecret, 3)
+
+	er, err := f.CRClient.EtcdV1beta2().EtcdRestores(f.Namespace).Create(er)
+	if err != nil {
+		t.Fatalf("failed to create etcd restore cr: %v", err)
+	}
+	defer func() {
+		if err := f.CRClient.EtcdV1beta2().EtcdRestores(f.Namespace).Delete(er.Name, nil); err != nil {
+			t.Fatalf("failed to delete etcd restore cr: %v", err)
+		}
+	}()
+
+	// Verify the EtcdRestore CR status "succeeded=true"
+	err = retryutil.Retry(time.Second, 5, func() (bool, error) {
```

Comment:
Can you write down what's the normal finished time?

Reply:
In practice, it usually takes about 1 second for `succeeded=true` from the time of the restore CR creation.
I can add a comment. Do you want to keep the retry time and attempts the same?

Reply:
Just write down the time is enough. In case of flake, there is reference to look at.